packages/next/src/generators/application/lib/add-linting.ts

```diff
@@ -39,9 +39,17 @@ export async function addLinting(
             `${options.appProjectRoot}/tsconfig(.*)?.json`,
           ];
         }
-        if (!reactEslintJson.extends) {
-          reactEslintJson.extends = [];
+        // Configure custom pages directory for next rule
+        if (reactEslintJson.overrides?.[0].rules) {
```

Comment:
It's an edge case, but technically there is nothing to guarantee that the first override block is the correct one, a user could have shifted things around from what is generated by Nx. It would be safer to check for the presence of the file extensions you can care about to locate the appropriate override block